### ISSUE TITLE
Improve stacktraces in logging

### ISSUE THREAD

**Issue by MoritzWeber0** [DSD-DBS/capella-collab-manager#577]
In our logs, the stacktraces should be improvement. When an error during a call occurs, there is no real stracktrace, just messages from fastAPI - this makes it hard to find the origin. 

How to reproduce: 
1. Add the following line as first line of the `/persistent` session route to:
    ```py
    this_should_fail()
    ```
2. Try starting a session
3. The log is: 
    ```py
    time="2023-02-01 08:35:41,194" level=ERROR function=dispatch client="127.0.0.1:50431" trace_id="FGTGLMXC8V" method="POST" path="/api/v1/sessions/persistent" user="aa88f8c7-868d-4fe6-bb2e-658f34c5c602" message="name 'this_should_fail' is not defined"
    Traceback (most recent call last):
      File "projectlocation/capella-collab-manager/backend/.venv/lib/python3.11/site-packages/anyio/streams/memory.py", line 94, in receive
        return self.receive_nowait()
              ^^^^^^^^^^^^^^^^^^^^^
      File "projectlocation/capella-collab-manager/backend/.venv/lib/python3.11/site-packages/anyio/streams/memory.py", line 89, in receive_nowait
        raise WouldBlock
    anyio.WouldBlock

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "projectlocation/capella-collab-manager/backend/.venv/lib/python3.11/site-packages/starlette/middleware/base.py", line 77, in call_next
        message = await recv_stream.receive()
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "projectlocation/capella-collab-manager/backend/.venv/lib/python3.11/site-packages/anyio/streams/memory.py", line 114, in receive
        raise EndOfStream
    anyio.EndOfStream

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "projectlocation/capella-collab-manager/backend/capellacollab/core/logging.py", line 77, in dispatch
        return await call_next(request)
              ^^^^^^^^^^^^^^^^^^^^^^^^
      File "projectlocation/capella-collab-manager/backend/.venv/lib/python3.11/site-packages/starlette/middleware/base.py", line 80, in call_next
        raise app_exc
      File "projectlocation/capella-collab-manager/backend/.venv/lib/python3.11/site-packages/starlette/middleware/base.py", line 69, in coro
        await self.app(scope, receive_or_disconnect, send_no_error)
      File "projectlocation/capella-collab-manager/backend/.venv/lib/python3.11/site-packages/starlette/middleware/base.py", line 106, in __call__
        response = await self.dispatch_func(request, call_next)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "projectlocation/capella-collab-manager/backend/capellacollab/core/logging.py", line 88, in dispatch
        response: Response = await call_next(request)
                            ^^^^^^^^^^^^^^^^^^^^^^^^
      File "projectlocation/capella-collab-manager/backend/.venv/lib/python3.11/site-packages/starlette/middleware/base.py", line 80, in call_next
        raise app_exc
      File "projectlocation/capella-collab-manager/backend/.venv/lib/python3.11/site-packages/starlette/middleware/base.py", line 69, in coro
        await self.app(scope, receive_or_disconnect, send_no_error)
      File "projectlocation/capella-collab-manager/backend/.venv/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
        raise exc
      File "projectlocation/capella-collab-manager/backend/.venv/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
        await self.app(scope, receive, sender)
      File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/contextlib.py", line 222, in __aexit__
        await self.gen.athrow(typ, value, traceback)
      File "projectlocation/capella-collab-manager/backend/.venv/lib/python3.11/site-packages/fastapi/concurrency.py", line 36, in contextmanager_in_threadpool
        raise e
    NameError: name 'this_should_fail' is not defined
    ```
4. The stacktrace has no context at all.

**Comment by MoritzWeber0** [comment]
This is a general `fastapi` issue: https://github.com/tiangolo/fastapi/issues/5740

**Comment by MoritzWeber0** [comment]
Fixed with Python 3.11.2: https://github.com/tiangolo/fastapi/discussions/8428#discussioncomment-5364481
